### PR TITLE
Address #1996

### PR DESCRIPTION
**Pull request by ben-arnao**
Let users know that weight decay optimizers are not compatible with SWA

### CODE REVIEW

File: tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py [hunk 1]
```diff
@@ -123,3 +123,20 @@ def test_serialization():
     config = tf.keras.optimizers.serialize(optimizer)
     new_optimizer = tf.keras.optimizers.deserialize(config)
     assert new_optimizer.get_config() == optimizer.get_config()
+    
+def test_weight_decay_incompatibility():
+    opt = SGD(learning_rate=1e-1)
```
Comment (bhack) [inline tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py:128]: `SGD`?

File: tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py [hunk 2]
```diff
@@ -123,3 +123,15 @@ def test_serialization():
...
     assert new_optimizer.get_config() == optimizer.get_config()
+    
+def test_weight_decay_incompatibility():
+    try:
+        SWA(tf.keras.optimizers.SGD())
+    except TypeError:
+        assert False
+
+    try:
+        SWA(tensorflow_addons.optimizers.SGDW())
```
Comment (bhack) [inline tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py:134]: tensorflow_addons?

Comment (ben-arnao) [reply tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py]: I hope i fixed it now

**Comment by bhack** [comment]
Please check https://github.com/tensorflow/addons/blob/master/STYLE_GUIDE.md#python

**Comment by seanpmorgan** [comment]
See the colab in the #1996:
https://colab.research.google.com/drive/1b-8JSEMx0Ruqfue9L092sCQxq7-KJaZW?usp=sharing
